### PR TITLE
lesson13 done

### PR DESCRIPTION
**Pull request by Naumych**
Added: 
-list of stations and trains with partials
- footer
- interface wor routes(it aactually was)
- adding possibility to add current_station and route to train
- 'show' from train have route of this train or 'no route'
- 'show' from route have list of trains or 'no trains'

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -10,6 +10,24 @@
...
+footer {
+    position: absolute;
+    bottom: 0;
+    width: 100%;
+    height: 10vh; /*высота футера будет зависеть от высоты экрана*/
+}
+/* чтобы весь футер заливался */
+footer .row{
+    background-color: #193441;
+}
```
Comment (svidlonix) [inline app/assets/stylesheets/application.css]: Please add the last empty line.

File: app/views/common/_errors.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<div id="error_explanation">
+    <h2><%= pluralize(resourse.errors.count, "error") %> prohibited this route from being saved:</h2>
+
+    <ul>
+    <% resourse.errors.full_messages.each do |message| %>
+    <li><%= message %></li>
+    <% end %>
+    </ul>
+</div>
+<% end %>
```
Comment (svidlonix) [inline app/views/common/_errors.html.erb]: Please add the last empty line.

File: app/views/railway_stations/_railway_station.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<tr>
+<td><%= railway_station.title %></td>
+<td><%= link_to 'Show', railway_station %></td>
+<td><%= link_to 'Edit', edit_railway_station_path(railway_station) %></td>
+<td><%= link_to 'Destroy', railway_station, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+</tr>
```
Comment (svidlonix) [inline app/views/railway_stations/_railway_station.html.erb]: Please add the last empty line.

File: app/assets/stylesheets/application.css [hunk 4]
```diff
@@ -10,6 +10,24 @@
  * files in this directory. Styles in this file should be added after the last require_* statement.
  * It is generally better to create a new file per style scope.
  *
- *= require_tree .
- *= require_self
  */
+
```
Comment (svidlonix) [inline app/assets/stylesheets/application.css]: Don't use 'application.css' for styles.

File: app/views/railway_stations/show.json.jbuilder [hunk 5]
```diff
@@ -1,3 +0,0 @@
-# frozen_string_literal: true
```
Comment (svidlonix) [inline app/views/railway_stations/show.json.jbuilder]: If you destroy this action https://github.com/Naumych/studybook_on_rails/blob/master/app/controllers/railway_stations_controller.rb#L46
why do you destroy view ?

File: app/views/routes/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit "#{ @route.new_record? ? 'Create' : 'Edit' }" %>
+  </div>
+<% end %>
```
Comment (svidlonix) [inline app/views/routes/_form.html.erb]: Please add the last empty line.

File: app/views/routes/_route.html.erb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<tr>
+<td><%= route.name %></td>
+<td><%= link_to 'Show', route %></td>
+<td><%= link_to 'Edit', edit_route_path(route) %></td>
+<td><%= link_to 'Destroy', route, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+</tr>
```
Comment (svidlonix) [inline app/views/routes/_route.html.erb]: Please add the last empty line.

File: app/views/trains/_form.html.erb [hunk 8]
```diff
@@ -1,21 +1,22 @@
...
+    <%= form.label :current_station, "Current station" %>
+    <%= form.collection_select :current_station_id, RailwayStation.all, :id, :title  %>
+  </div>
+
+  <div class="field">
+    <%= form.label :route, "Current route" %>
+    <%= form.collection_select :route_id, Route.all, :id, :name  %>
+  </div>
+
+
```
Comment (svidlonix) [inline app/views/trains/_form.html.erb]: please delete the extra line.

File: app/views/trains/_train.html.erb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<tr>
+<td><%= train.number %></td>
+<td><%= link_to 'Show', train %></td>
+<td><%= link_to 'Edit', edit_train_path(train) %></td>
+<td><%= link_to 'Destroy', train, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+</tr>
```
Comment (svidlonix) [inline app/views/trains/_train.html.erb]: Please add the last empty line.

**Comment by svidlonix** [comment]
Looks like bootstrap doesn't work in your project, try please again to connect it.

File: app/assets/stylesheets/application.css (hunk 4 above)
Comment (Naumych) [reply app/assets/stylesheets/application.css]: @svidlonix  I don't know what should I use

**Comment by svidlonix** [comment]
Please fix bug https://i.imgur.com/8kmH0Cu.png

File: app/views/common/_errors.html.erb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<% if resourse.errors.any? %>
```
Comment (svidlonix) [inline app/views/common/_errors.html.erb]: Please add indent

File: app/views/routes/show.html.erb [hunk 11]
```diff
@@ -4,6 +4,18 @@
   <strong>Name:</strong>
   <%= @route.name %>
 </p>
+<p>
+  <strong>Trains on the route: </strong>
+  <% if @route.trains.present? %>
+    <ul>
+    <% @route.trains.each do |train| %>
```
Comment (svidlonix) [inline app/views/routes/show.html.erb]: Please add indent

File: app/views/trains/_train.html.erb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<tr>
+<td><%= train.number %></td>
```
Comment (svidlonix) [inline app/views/trains/_train.html.erb]: Please add indent

**Pull request merged**